src/sqlfluff/core/parser/segments/base.py

```diff
@@ -719,11 +719,7 @@ def path_to(self, other):
             return [self]
 
         # Are we in the right ballpark?
-        if (
-            not self.get_start_point_marker()
-            <= other.get_start_point_marker()
-            <= self.get_end_point_marker()
-        ):
+        if not self.get_start_loc() <= other.get_start_loc() <= self.get_end_loc():
```

Comment:
Does removing the parentheses affect the logic here? I find myself asking, which is higher precedence, `not` or the range comparison operators? 😅

Nitpick: Could it be worthwhile to restructure the function like this to avoid the slightly hard-to-read `not` structure?
```
if self.get_start_loc() <= other.get_start_loc() <= self.get_end_loc():
        # Do we have any child segments at all?
        if not self.segments:
            return None
else:
    # This code was formerly under the `if not ...`
    return None
```

Reply:
Good idea.

Reply:
Actually, I'm going to leave the code as is (partly because black says so), and because it is clearer provided you know the precedence. I've added a comment to explain that though.